src/org/zaproxy/zap/extension/csphelper/CspElement.java

```diff
@@ -0,0 +1,263 @@
+import java.util.TreeSet;
+
+import org.apache.log4j.Logger;
+
+public class CspElement {
+
+	private CSP csp;
+	private String name;
+	private boolean enabled;
+	
+	private boolean none;
+	private boolean all;
+	private boolean self;
+	private boolean data;
+	private boolean mediastream;
+	private boolean blob;
+	private boolean filesystem;
+	private boolean unsafeInline;
+	private boolean unsafeEval;
+	private boolean strictDynamic;
+	private boolean unsafeHashedAttr;
+	private boolean noHashNonce;
+	private Set<String> extras = new TreeSet<String>();
+	private Set<String> hashNonces = new TreeSet<String>();
+
+	private static final Logger LOGGER = Logger.getLogger(CspElement.class);
+
+	public CspElement (CSP csp, String name) {
+		this.csp = csp;
+		this.name = name;
+	}
+	
+	public String getName() {
+		return this.name;
+	}
+	
+	protected void reset() {
+		this.enabled = false;
+		this.none = false;
+		this.all = false;
+		this.self = false;
+		this.data = false;
+		this.noHashNonce = false;
+		this.extras.clear();
+		this.hashNonces.clear();
+	}
+	
+	public boolean getEnabled() {
+		return this.enabled;
+	}
+	
+	public void setEnabled() {
+		reset();
+		this.enabled = true;
+	}
+	
+	public void setNone() {
+		reset();
+		this.none = true;
+		this.enabled = true;
+	}
+	
+	public void setAll() {
+		reset();
+		this.all = true;
+		this.enabled = true;
+	}
+	
+	public void setSelf() {
+		this.none = false;
+		this.all = false;
+		this.self = true;
+		this.enabled = true;
+	}
+	
+	public void setData() {
+		this.none = false;
+		this.all = false;
+		this.data = true;
+		this.enabled = true;
+	}
+	
+	public void setMediastream() {
+		this.none = false;
+		this.all = false;
+		this.mediastream = true;
+		this.enabled = true;
+	}
+	
+	public void setBlob() {
+		this.none = false;
+		this.all = false;
+		this.blob = true;
+		this.enabled = true;
+	}
+	
+	public void setFilesystem() {
+		this.none = false;
+		this.all = false;
+		this.filesystem = true;
+		this.enabled = true;
+	}
+	
+	public void setUnsafeInline() {
+		this.none = false;
+		this.all = false;
+		this.unsafeInline = true;
+		this.noHashNonce = true;
+		this.enabled = true;
+	}
+	
+	public void setUnsafeEval() {
+		this.none = false;
+		this.all = false;
+		this.unsafeEval = true;
+		this.enabled = true;
+	}
+	
+	public void setStrictDynamic() {
+		this.none = false;
+		this.all = false;
+		this.strictDynamic = true;
+		this.enabled = true;
+	}
+	
+	public void setUnsafeHashedAttribute() {
+		this.none = false;
+		this.all = false;
+		this.unsafeHashedAttr = true;
+		this.enabled = true;
+	}
+	
+	public void addUrl(String url) {
+		this.none = false;
+        this.enabled = true;
+        url = url.trim();
+        if ((url.startsWith("http://") || url.startsWith("https://"))) {
+        	url = ExtensionCspHelper.siteFromUrl(url);
+        }
+		if (url.startsWith(csp.getSite())) {
+			LOGGER.debug("element " + name + " addUrl " + url + " isSelf (starts with site)");
```

Comment:
All debug with concatenation should be wrapped in an isDebugEnabled check....throughout this PR